pyscripts/vc/rc_path_finder.py

```diff
@@ -0,0 +1,54 @@
+import os
+import re
+
+
+def create_build_spec_coord2path_dic(repo_path):
+    with open(os.path.join(repo_path, 'README.md'), 'r') as file:
+        content = file.read()
+
+    table_content = re.search(r"<!-- BEGIN GENERATED RESULTS TABLE -->(.*?)<!-- END GENERATED RESULTS TABLE -->",
+                              content, re.DOTALL).group(1)
+    table_lines = table_content.strip().split('\n')[2:-1]  # Skip table headers and last row stats
+
+    artifacts_dict = {}
+    for line in table_lines:
+        matches = re.search(r"\| ?(\S*?) ?\| <a name='(\S+?:\S+?)'></a>\[(\S+?)\]\((\S+?)\)", line)
+        group, full_name, artifact, link = matches.groups()
+        modules = {full_name}
+        if link:
+            artifact_readme_path = os.path.join(repo_path, link)
+            with open(artifact_readme_path, 'r') as file:
+                content = file.read()
+            table_pattern = r"<details><summary>This project defines \d+ modules:</summary>(.*?)</details>"
+            table = re.search(table_pattern, content, re.DOTALL)
+
+            if table is not None:
+                artifact_pattern = r"\[\s*([\w\.-]+:[\w\.-]+)\s*\]"
+                artifact_matches = re.findall(artifact_pattern, table.group(1))
+                modules.update(set(artifact_matches))
+
+        versions = re.findall(r'\[\d+\.\d+\.\d+\]', content)
```

Comment:
If I am not mistaken, this captures only versions in the format [x.y.z]. However, versions like 1, 1.0, 1.0.2-RC1 or 1.0.FINAL, etc would not be matched. For example [this package](https://github.com/jvm-repo-rebuild/reproducible-central/blob/master/content/org/apache/felix/http/parent/README.md) has version=13.  I think the table itself should be parsed and all versions extracted, or use a different regex.

Reply:
Good catch, thanks for pointing it out. I improved the regex so that we cover such cases. Now the number of all coordinates we can find is 107,304 and 42,049 of them have buildspec.